### PR TITLE
[FIX] app: make subroots more robust

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

**Review by pparidans — APPROVED** [review]
LGTM

File: tests/app/sub_root.test.ts [hunk 1]
```diff
@@ -112,4 +112,66 @@ describe("subroot", () => {
...
+
+  const app = new App(SomeComponent);
+  const comp = await app.mount(fixture);
+  expect(fixture.innerHTML).toBe("a<div></div>");
+  await nextTick();
+  expect(fixture.innerHTML).toBe("a<div>c</div>");
+  comp.state.flag = true;
+  await nextTick();
+  expect(fixture.innerHTML).toBe("b");
+  app.createRoot(SubComponent);
```
Comment (rfr-odoo) [inline tests/app/sub_root.test.ts:176]: Is needed as there is no `expect` after this line or its to test that will not crash ?

Comment (ged-odoo) [reply tests/app/sub_root.test.ts:176]: hmm, good question. looking into it

Comment (pparidans) [reply tests/app/sub_root.test.ts]: We call him "The eagle eye" for a reason... :wink:

**Review by rfr-odoo — APPROVED** [review]

**Pull request merged**